wasm/text/type_use_parser.go

```diff
@@ -0,0 +1,465 @@
+package text
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/tetratelabs/wazero/wasm"
+)
+
+func newTypeUseParser(typeNamespace *indexNamespace) *typeUseParser {
+	return &typeUseParser{typeNamespace: typeNamespace}
+}
+
+// typeUseParser parses an inlined type from a field such "func" and dispatches to onTypeUseEnd or onUnknownField.
+//
+// Ex. `(import "Math" "PI" (func $math.pi (result f32))`
+//                           starts here --^           ^
+//                         onTypeUseEnd resumes here --+
+//
+// Note: Unlike normal parsers, this is not used for an entire field (enclosed by parens). Rather, this only handles
+// "type", "param" and "result" inner fields in the correct order.
+// Note: typeUseParser is reusable. The caller resets via begin.
+type typeUseParser struct {
```

Comment:
this no longer has code inside to parse module-defined types, which allows it to be smaller than it was before. It was getting out-of-hand.